frontend/src/views/Game.vue

```diff
@@ -1,14 +1,166 @@
+    document.removeEventListener('keyup', this.handleKeyUp);
+    this.socketioGame.disconnect();
+  },
+  methods: {
+    runGame() {
+      this.inGame = true;
+      this.socketioGame.setupSocketConnection(this.$cookie.getCookie('jwt'));
+      this.socketioGame.subscribe('gameLoop', (datas) => {
+        this.ball = datas.ball;
+        this.pad1 = datas.pad1;
+        this.pad2 = datas.pad2;
+        this.map = datas.map;
+        this.score = datas.score;
+        this.draw();
+      });
+      this.socketioGame.subscribe('gameOver', (res) => {
+        document.removeEventListener('keydown', this.handleKeyDown);
+        document.removeEventListener('keyup', this.handleKeyUp);
+        this.canvas = null;
+        this.ctx = null;
+        const score = res.score;
+        this.winnerId =
+          score.player1.points > score.player2.points
+            ? score.player1.id
+            : score.player2.id;
+      });
+      this.init();
+      //this.sendGameDatas();
+      this.socketioGame.send('connectToRoom');
+      document.addEventListener('keydown', this.handleKeyDown);
+      document.addEventListener('keyup', this.handleKeyUp);
+    },
+    // function for custom games
+    sendGameDatas() {
+      const gameDatas = {
+        map: this.map,
+        ball: this.ball,
+        pad1: this.pad1,
+        pad2: this.pad2,
+        score: this.score
+      };
+      this.socketioGame.send('initGame', gameDatas);
+    },
+    init() {
+      this.canvas = document.getElementById('canvas');
+      this.ctx = this.canvas.getContext('2d');
+    },
+    draw() {
+      if (!this.map) return;
+      this.ctx.clearRect(0, 0, this.map.width, this.map.height);
+      this.writeScore();
+      this.createCenterDotLine();
+      this.drawBall();
+      this.drawPad(this.pad1);
+      this.drawPad(this.pad2);
+    },
+    createCenterDotLine() {
+      this.ctx.beginPath();
+      this.ctx.setLineDash([5, 15]);
+      this.ctx.moveTo(this.map.width / 2, 0);
+      this.ctx.lineTo(this.map.width / 2, this.map.height);
+      this.ctx.strokeStyle = 'white';
+      this.ctx.stroke();
+    },
+    drawBall() {
+      this.ctx.beginPath();
+      this.ctx.arc(
+        this.ball.x,
+        this.ball.y,
+        this.ball.radius,
+        0,
+        Math.PI * 2,
+        true
+      );
+      this.ctx.closePath();
+      this.ctx.fillStyle = 'white';
+      this.ctx.fill();
+    },
+    drawPad(pad) {
+      this.ctx.beginPath();
+      this.ctx.rect(pad.x, pad.y, pad.width, pad.height);
+      this.ctx.closePath();
+      this.ctx.fillStyle = 'white';
+      this.ctx.fill();
+    },
+    writeScore() {
+      this.ctx.font = '20px Poppins';
+      this.ctx.fillStyle = 'white';
+      this.ctx.fillText(this.score.player1.points, this.map.width / 2 - 25, 20);
+      this.ctx.fillText(this.score.player2.points, this.map.width / 2 + 15, 20);
+    },
+    handleKeyDown(e) {
+      if (e.keyCode === 38) {
```

Comment:
Please name the keycode instead of using a magic number :3

Reply:
Magic is cool